### PR TITLE
Service navigation component

### PR DESCRIPTION
**Pull request by querkmachine**
This PR supersedes #4950. This PR should be considered the working branch for the navigation work.

I opted to make a new PR as we've decided to remove or refactor many aspects of the previous PR. However, we may still integrate those aspects at a later date, so having the previous PR exist as-is as a point of reference and comparison is still useful.

## Changes from the previous PR

- [x] Rename component from Service Header to Service Navigation
- [x] ~~Refactor navigation links so active styles applied directly to link/span, rather than the list item~~
  - We ultimately decided to revert this change and continue using the previous active style handling
- [x] Refactor class names to avoid the new `navigation__navigation` repetition
- [x] Make navigation items mandatory, rework examples/tests to have nav items as default
- [x] Darken link colour by 10% to increase perceptual contrast
- [x] Remove the `<section>` element if a service name isn't present
- [x] Undo changes and remove slots from the Header component
  - We later readded the full-width bottom border as a variant of the component

### CODE REVIEW

**Review by owenatgov — COMMENTED** [review]
A few minor comments. We're looking good. I've got a couple of extra points to make outside code. One thing I think we should do before the launch party and 3 things I'm willing to negotiate on and park for later:

## Gotta do
The commit history could do with a tidy. I suspect this was on your radar already but just to enshrine it as a thing we should do.

## Don't gotta do now
Firstly, I think an example of the service nav would be good. Either replacing the header nav in an existing example or creating a new one showing the header and the service nav together. The latter would be my preference.

Secondly, there's currently no way to translate the service nav. This can be an iteration but it's a common pattern in our components so we should do it eventually.

Finally, I noticed on mobile that the margin collapse isn't working between the service name and the menu button, creating a very slightly jarring, bigger than expected space:
40px space between service name and menu button

I know why this is happening: the margin for the button is applied inside the nav element and it doesn't bubble out to its parent. This can be solved with a bit of margin wrangling on mobile which is annoying but doable. I don't know if it's intentional and if it isn't it's not a reason not to merge, it just looks a bit weird to my eye.

**Comment by querkmachine** [comment]
@owenatgov Which parts cannot be translated? I can't see anything hardcoded at a glance.

File: packages/govuk-frontend/src/govuk/components/service-navigation/_index.scss [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    @include govuk-media-query($from: tablet) {
+      padding-bottom: govuk-spacing(4) - $govuk-service-navigation-active-link-border-width;
+      border-bottom-width: $govuk-service-navigation-active-link-border-width;
+    }
+  }
+
+  // These styles are shared by links and non-linked text items
+  .govuk-service-navigation__link,
+  .govuk-service-navigation__text {
+    display: inline-block;
```
[the remark this answers is not in the document]
Comment (querkmachine) [reply packages/govuk-frontend/src/govuk/components/service-navigation/_index.scss:63]: Methinks think this is a layover from when we refactored and then reverted how we were styling links.

File: packages/govuk-frontend/src/govuk/components/service-navigation/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Service Navigation
+
+## Installation
+
+See the [main README quick start guide](https://github.com/alphagov/govuk-frontend#quick-start) for how to install this component.
+
+## Guidance and Examples
```
[the remark this answers is not in the document]
Comment (querkmachine) [reply packages/govuk-frontend/src/govuk/components/service-navigation/README.md:7]: I've left this one as-is, for as weird as it is, it's a capitalisation we use [in the README files for every component](https://github.com/search?q=repo%3Aalphagov%2Fgovuk-frontend%20path%3A%2F(%5E%7C%5C%2F)README%5C.md%24%2F%20%22Guidance%20and%20Examples%22&type=code) and should probably be dealt with systematically.

**Comment by owenatgov** [comment]
@querkmachine Ah whoops you're right! Sorry, habitual reviewing.

**Review by owenatgov — APPROVED** [review]
Code is looking _sharp_. Congratulations and well done ✨

I'm going to pop this note here that I unfortunately noticed another example of non-collapsing margin 🥲 This time at tablet when the nav itself overflows:

Big space between service name and nav on big screens

This one looks less weird and I suspect is even more annoying to fix so I don't think it's worth busting our backs to get rid of it given the time frames. Maybe worth reviewing post release when we've got a bit more time.

**Comment by querkmachine** [comment]
Yeah, that one is a lot trickier as the wrapping is a result of `flex-wrap`, which in turn happens (or doesn't happen) depending on the length of the service name and navigation items. It's not a case of being able to tweak styles for a specific breakpoint.

File: packages/govuk-frontend/src/govuk/components/service-navigation/README.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# Service navigation
+
+## Installation
+
+See the [main README quick start guide](https://github.com/alphagov/govuk-frontend#quick-start) for how to install this component.
+
+## Guidance and Examples
+
+Find out when to use the service header component in your service in the [GOV.UK Design System](https://design-system.service.gov.uk/components/service-navigation).
```
[the remark this answers is not in the document]
Comment (querkmachine) [reply packages/govuk-frontend/src/govuk/components/service-navigation/README.md]: Nope! That's the component's old name. Updated to say service navigation.

**Pull request merged**